src/csharp/Grpc.Core/RpcException.cs

```diff
@@ -44,6 +47,18 @@ public RpcException(Status status) : base(status.ToString())
         public RpcException(Status status, string message) : base(message)
         {
             this.status = status;
+            this.trailers = Metadata.Empty;
+        }
+
+        /// <summary>
+        /// Creates a new <c>RpcException</c> associated with given status and trailing response metadata.
+        /// </summary>
+        /// <param name="status">Resulting status of a call.</param>
+        /// <param name="trailers">Response trailing metadata.</param> 
+        public RpcException(Status status, Metadata trailers) : base(status.ToString())
```

Comment:
I thought of that, but the problem is that making constructor private might make it hard to write tests in projects that use gRPC - essentially it means it's impossible to construct an instance of RpcException that contains trailers and that's something you might well want to do in your tests (we've had complaints about internal/private constructors in other parts of our API already, so I don't want to repeat the same mistake twice).
Also, there's some added value in being able to add trailers by throwing RpcException in server-side handlers.